script/foreman-debug

```diff
@@ -79,41 +80,52 @@ add_cmd() {
   eval $CMD >> "$DIR/$OUT" 2>&1
 }
 
-# add and filter if it is a non zero, readable, regular file or symlink (skip otherwise)
-add_files() {
-  if [ $MAXLINES -eq 0 ]; then
-    TAIL_OR_CAT="cat"
-  else
-    TAIL_OR_CAT="tail -n ${MAXLINES}"
-  fi
-  for FILE in $*; do
-    if [ \( -f "$FILE" -o -h "$FILE" \) -a \( -r "$FILE" -a -s "$FILE" \) ]; then
-      printv " - $FILE"
-      SUBDIR=$(dirname $FILE)
-      [ ! -d "$DIR$SUBDIR" ] && mkdir -p "$DIR$SUBDIR"
-      MIME=$(file -bi "$FILE" | cut -d\; -f1)
-      case $MIME in
+# add and filter file of known MIME type, calculate size
+add_file() {
+  FILE=$1
+  SIZE=$(du -k $FILE | cut -f1)
+  MIME=$(file -bi "$FILE" | cut -d\; -f1)
+  case $MIME in
         application/x-gzip)
-          zcat "$FILE" | $TAIL_OR_CAT | sed -r "$FILTER" > "$DIR$FILE.txt"
+          zcat "$FILE" | sed -r "$FILTER" > "$DIR$FILE.txt"
           ;;
         application/x-bzip2)
-          bzcat "$FILE" | $TAIL_OR_CAT | sed -r "$FILTER" > "$DIR$FILE.txt"
+          bzcat "$FILE" | sed -r "$FILTER" > "$DIR$FILE.txt"
           ;;
         application/x-xz)
-          xzcat "$FILE" | $TAIL_OR_CAT | sed -r "$FILTER" > "$DIR$FILE.txt"
+          xzcat "$FILE" | sed -r "$FILTER" > "$DIR$FILE.txt"
           ;;
         text/plain | application/xml)
-          $TAIL_OR_CAT "$FILE" | sed -r "$FILTER" > "$DIR$FILE"
+          sed -r "$FILTER" "$FILE" > "$DIR$FILE"
           [ $PRINTPASS -eq 1 ] && grep -H "\*\*\*\*\*" "$DIR$FILE"
           ;;
-        inode/symlink)
-          cp -a "$FILE" "$DIR$FILE"
-          add_files $(readlink -f $FILE)
-          ;;
         *)
           echo "Skipping file $FILE: unknown MIME type $MIME" >> "$DIR/skipped_files"
+          SIZE=0 # don't count the size to collected files
           ;;
       esac
+}
+
+# add files (from newest to oldest) that are non zero, readable, regular file or symlink, until $MAXSIZE limit is reached
+add_files() {
+  SUMSIZE=0
+  SKIP="" # will be nonempty to stop adding files
+  # sort regular+symlink readable nonempty files per modification time, newest first - assuming no space in a filename
+  for FILE in $(find -L $* -type f -readable -size +0b -printf "%T@ %p\n" 2> /dev/null | sort -nr | cut -d' ' -f2 2> /dev/null); do
+    # if we are over size limit, skip rest older files
+    if [ \( $SUMSIZE -gt $MAXSIZE \) -a \( $MAXSIZE -gt 0 \) ]; then
```

Comment:
I think the `\( $MAXSIZE -gt 0 \)` test should be done outside of the loop, perhaps just make sure the user entered a positive regular number and that's it.